services/src/handlers/layers.rs

```diff
@@ -455,3 +485,634 @@ async fn layer_handler<C: Context>(
+    pub collection_id: LayerCollectionId,
+}
+
+/// Add a new layer to a collection
+#[utoipa::path(
+    tag = "Layers",
+    post,
+    path = "/layerDb/layers",
+    request_body = AddLayerRequest,
+    responses(
+        (status = 200, description = "OK", body = IdResponse<LayerId>,
+            example = json!({
+                "id": "36574dc3-560a-4b09-9d22-d5945f2b8093"
+            })
+        )
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn add_layer<C: Context>(
+    _session: AdminSession, // TODO: allow normal users to add layers to their stuff
+    ctx: web::Data<C>,
+    request: web::Json<AddLayerRequest>,
+) -> Result<web::Json<IdResponse<LayerId>>> {
+    let request = request.into_inner();
+
+    let collection_id = request.collection_id;
+    let add_layer = request.layer.validated()?;
+
+    let id = ctx
+        .layer_db_ref()
+        .add_layer(add_layer, &collection_id)
+        .await?;
+
+    Ok(web::Json(IdResponse { id }))
+}
+
+/// Add a new collection to an existing collection
+#[utoipa::path(
+    tag = "Layers",
+    post,
+    path = "/layerDb/collections/{collection}/collections",
+    params(
+        ("collection" = LayerCollectionId, description = "Layer collection id", example = "05102bb3-a855-4a37-8a8a-30026a91fef1"),
+    ),
+    request_body = AddLayerCollection,
+    responses(
+        (status = 200, description = "OK", body = IdResponse<LayerCollectionId>,
+            example = json!({
+                "id": "36574dc3-560a-4b09-9d22-d5945f2b8093"
+            })
+        )
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn add_collection<C: Context>(
+    _session: AdminSession, // TODO: allow normal users to add collections to their stuff
+    ctx: web::Data<C>,
+    collection: web::Path<LayerCollectionId>,
+    request: web::Json<AddLayerCollection>,
+) -> Result<web::Json<IdResponse<LayerCollectionId>>> {
+    let add_collection = request.into_inner().validated()?;
+
+    let id = ctx
+        .layer_db_ref()
+        .add_collection(add_collection, &collection)
+        .await?;
+
+    Ok(web::Json(IdResponse { id }))
+}
+
+/// Remove a collection
+#[utoipa::path(
+    tag = "Layers",
+    delete,
+    path = "/layerDb/collections/{collection}",
+    params(
+        ("collection" = LayerCollectionId, description = "Layer collection id"),
+    ),
+    responses(
+        (status = 200, description = "OK")
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn remove_collection<C: Context>(
+    _session: AdminSession, // TODO: allow normal users to remove their collections
+    ctx: web::Data<C>,
+    collection: web::Path<LayerCollectionId>,
+) -> Result<HttpResponse> {
+    ctx.layer_db_ref().remove_collection(&collection).await?;
+
+    Ok(HttpResponse::Ok().finish())
+}
+
+#[derive(Debug, Serialize, Deserialize, IntoParams)]
+struct RemoveLayerFromCollectionParams {
+    collection: LayerCollectionId,
```

Comment:
while in general (in the layer API) a id can be an arbitrary string (necessary because we have to support arbitrary external providers), for the layerDb we know they all have to be UUIDs. Maybe just add a TODO to reflect that in the OpenAPI spec?

Reply:
[9b64131](https://github.com/geo-engine/geoengine/pull/676/commits/9b64131ef6a6b7b53ebb007e025ec12e559363e6)